dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -16,37 +16,205 @@
 
 package com.palantir.dialogue;
 
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.ListMultimap;
 import com.google.common.collect.Multimap;
+import com.palantir.logsafe.Preconditions;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
-import org.immutables.value.Value;
 
 /** Defines the parameters of a single {@link Call} to an {@link Endpoint}. */
-@DialogueImmutablesStyle
-@Value.Immutable
-public interface Request {
-    /** The HTTP headers for this request, encoded as a map of {@code header-name: header-value}. */
-    Map<String, String> headerParams();
+public final class Request {
+
+    private final ImmutableSortedMap<String, String> headerParams;
```

Comment:
Although conjure does enforce a single value per header, I don't see a reason why the underlying client should be so strict. I think being able to to express any API (including non-conjurized ones) within dialogue is important if we want to build a full replacement for CJR since it currently supports any valid JAXRS interface.

As a concrete example, Github allows you to opt into API previews using multiple `Accept` headers

Reply:
Updated, I've also filed #343 to update the Response API to use a ListMultimap, but I'd prefer not to do that in this PR because it targets the Request side.

Regarding CJR, once dialogue is in a good place I'd like to replace our feign client with a dialogue channel as it should capture what we've tried to do with okhttp nicely.